### PR TITLE
Add support for APIM 4.1.0-beta

### PR DESCRIPTION
**Pull request by PasanT9**

### CODE REVIEW

File: docker-compose/apim-with-analytics/conf/apim/repository/conf/deployment.toml [hunk 1]
```diff
@@ -251,3 +255,8 @@ notification_endpoint = "https://localhost:${mgt.transport.https.port}/internal/
 username = "${admin.username}"
 password = "${admin.password}"
 'header.X-WSO2-KEY-MANAGER' = "default"
+
+[oauth.grant_type.token_exchange]
+enable = true
+allow_refresh_tokens = true
+iat_validity_period = "1h"
```
Comment (RakhithaRR) [inline docker-compose/apim-with-analytics/conf/apim/repository/conf/deployment.toml:262]: Add EOF newline. Check other places as well

File: docker-compose/apim-with-analytics/dockerfiles/apim/Dockerfile [hunk 2]
```diff
@@ -17,11 +17,13 @@
...
+FROM wso2am:4.1.0-ubuntu
 LABEL maintainer="WSO2 Docker Maintainers <dev@wso2.org>"
 
 # build arguments for external artifacts
 ARG MYSQL_CONNECTOR_VERSION=8.0.17
 
 # add MySQL JDBC connector to server home as a third party library
 ADD --chown=wso2carbon:wso2 https://repo1.maven.org/maven2/mysql/mysql-connector-java/${MYSQL_CONNECTOR_VERSION}/mysql-connector-java-${MYSQL_CONNECTOR_VERSION}.jar ${WSO2_SERVER_HOME}/repository/components/dropins/
+
+ADD --chown=wso2carbon:wso2 https://repo1.maven.org/maven2/mysql/mysql-connector-java/${MYSQL_CONNECTOR_VERSION}/mysql-connector-java-${MYSQL_CONNECTOR_VERSION}.jar ${WSO2_SERVER_HOME}/repository/components/lib/
```
Comment (RakhithaRR) [inline docker-compose/apim-with-analytics/dockerfiles/apim/Dockerfile:29]: We don't need to add this twice. Adding this to one of these places will be enough.

**Review by RakhithaRR — APPROVED** [review]

File: docker-compose/apim-with-analytics/conf/apim/repository/conf/deployment.toml [hunk 3]
```diff
@@ -95,9 +99,9 @@ gateway_labels =["Default"]
 #[apim.cache.tags]
 #expiry_time = "2m"
 
-[apim.analytics]
-enable = true
-config_endpoint = "https://analytics-event-auth.choreo.dev/auth/v1"
+[apim.analytics]
```
Comment (sajinieKavindya) [inline docker-compose/apim-with-analytics/conf/apim/repository/conf/deployment.toml:102]: Shall we remove these extra spaces?

Comment (PasanT9) [reply docker-compose/apim-with-analytics/conf/apim/repository/conf/deployment.toml:102]: Fixed with https://github.com/wso2/docker-apim/pull/472

**Pull request merged**